src/main/java/uk/nhs/digital/nhsconnect/lab/results/inbound/RecepProducerService.java

```diff
@@ -0,0 +1,151 @@
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Segment;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.EdifactValidationException;
+import uk.nhs.digital.nhsconnect.lab.results.sequence.SequenceService;
+import uk.nhs.digital.nhsconnect.lab.results.utils.TimestampService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
+public class RecepProducerService {
+
+    private final SequenceService sequenceService;
+    private final TimestampService timestampService;
+
+    public String produceRecep(final Interchange receivedInterchangeFromHa) throws EdifactValidationException {
+        final var segments = mapEdifactToRecep(receivedInterchangeFromHa);
+        return toEdifact(segments);
+    }
+
+    private String toEdifact(final List<Segment> segments) {
+        return segments.stream()
+            .map(Segment::toEdifact)
+            .collect(Collectors.joining("\n"));
+    }
+
+    private List<Segment> mapEdifactToRecep(final Interchange receivedInterchange) throws EdifactValidationException {
+        final List<Segment> segments = new ArrayList<>();
+
+        final var interchangeHeader = mapToInterchangeHeader(receivedInterchange);
+        final var messageHeader = new RecepMessageHeader();
+        segments.add(interchangeHeader);
+        segments.add(messageHeader);
+        final var beginningOfMessage = new RecepBeginningOfMessage();
+        segments.add(beginningOfMessage);
+        segments.add(mapToNameAndAddress(receivedInterchange));
+        final var translationDateTime = emptyRecepTimestamp();
+        segments.add(translationDateTime);
+        segments.addAll(mapToReferenceMessageReceps(receivedInterchange));
+        segments.add(mapToReferenceInterchange(receivedInterchange));
+        final var messageTrailer = new MessageTrailer(0);
+        final var interchangeTrailer = mapToInterchangeTrailer(receivedInterchange);
+        segments.add(messageTrailer);
+        segments.add(interchangeTrailer);
+
+        segments.forEach(Segment::preValidate);
+
+        setSequenceNumbers(segments, interchangeHeader, messageHeader, messageTrailer, interchangeTrailer);
+        setTimestamps(interchangeHeader, beginningOfMessage, translationDateTime);
+
+        segments.forEach(Segment::validate);
+
+        return segments;
+    }
+
+    private List<ReferenceMessageRecep> mapToReferenceMessageReceps(final Interchange receivedInterchange) {
+        return receivedInterchange.getMessages().stream()
+            .map(this::mapToReferenceMessage)
+            .collect(Collectors.toList());
+    }
+
+    private void setTimestamps(
+        final RecepHeader recepInterchangeHeader,
+        final RecepBeginningOfMessage recepBeginningOfMessage,
+        final RecepMessageDateTime recepTranslationDateTime
+    ) {
```

Comment:
This formatting doesn't sit right for me, just a suggestion:
```suggestion
    private void setTimestamps(final RecepHeader recepInterchangeHeader, final RecepBeginningOfMessage recepBeginningOfMessage,
                               final RecepMessageDateTime recepTranslationDateTime) {
```

Reply:
This is how I prefer to format methods with many parameters, as I feel it makes it easier to 1) see the items in the parameter list clearly and 2) see where the parameters stop and the method body begins. (Point 2 is relevant only when the final `) {` is kept on the last parameter and the indentation is the same for parameters and the body.)

Let's get more feedback from the team on opinions here; if there's a strong preference one way or the other then we'll go with that, but if the team feels it doesn't matter much then I'll keep my own preferred style.

To make it clear what the options are, I'll add code blocks here:

```
    private void setTimestamps(
        final RecepHeader recepInterchangeHeader,
        final RecepBeginningOfMessage recepBeginningOfMessage,
        final RecepMessageDateTime recepTranslationDateTime
    ) {
        doStuff();
    }
```
or
```
    private void setTimestamps(final RecepHeader recepInterchangeHeader, final RecepBeginningOfMessage recepBeginningOfMessage,
                               final RecepMessageDateTime recepTranslationDateTime) {
        doStuff();
    }
```
and the style I've seen before that I feel makes it difficult to see where the parameters end and the method starts:
```
    private void setTimestamps(
        final RecepHeader recepInterchangeHeader,
        final RecepBeginningOfMessage recepBeginningOfMessage,
        final RecepMessageDateTime recepTranslationDateTime) {
        doStuff();
    }
```

Reply:
I prefer the second style, but I don't have a strong preference for it as long as it's understandable where parameters stop and the method body begins

```
    private void setTimestamps(final RecepHeader recepInterchangeHeader, final RecepBeginningOfMessage recepBeginningOfMessage,
                               final RecepMessageDateTime recepTranslationDateTime) {
        doStuff();
    }
```

Reply:
Second one looks better for me as well

Reply:
```
private void setTimestamps(
    final RecepHeader recepInterchangeHeader,
    final RecepBeginningOfMessage recepBeginningOfMessage,
    final RecepMessageDateTime recepTranslationDateTime
) {
    doStuff();
}
```

This is the formatting I prefer and I've seen on other projects. More readable to me. I don't like methods stretching so much they go off the screen.
But I'm happy as long as we're consistent. This may be something worth adding to the Java Coding Standards on the project, https://gpitbjss.atlassian.net/wiki/spaces/NIA/pages/2108522539/Java+Coding+Standards

Reply:
3 to 2 in favour of the suggestion. Democracy!